packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/utils/findAllAndReplace.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @typedef Attribute
+ * @type { { type: string; repeatable: boolean }}
+ *
+ * @typedef Attributes
+ * @type {{ [key: string]: Attribute }}
+ *
+ * @typedef ReplacementFn
+ * @type {<TData extends object, TKey extends string>(data: TData[TKey], { path: string[]; parent: TData }) => any}
+ */
+
+/**
+ * This function will recursively find everything and replace it with a value
+ * based on the boolean return of the predicate function e.g. `type === 'relation'`.
+ *
+ * If you provide a function it will call that function with data value you're replacing with
+ * a second argument with the path to the value and it's parent.
+ *
+ * It's original use was for the preperation of action items for the INIT_FORM action. It requires
+ * knowledge of the `components` in the entity, however `components` doesn't change nor does the predicate
+ * function so we don't need to pass it everytime hence why it's curried.
+ *
+ * @type {<TData extends object = object>(data: { [key: string]: { attributes: Attributes } }, predicate?: (value: Attribute, { path: string[]; parent: TData }) => boolean, replacement?: ReplacementFn<TData, keyof TKey> | any) => (data: TData, attributes: Attributes) => TData}
+ */
+const findAllAndReplaceSetup = (components, predicate = () => false, replacement = undefined) => {
+  /**
+   * @type {<TData extends object = object>(data: TData, attributes: Attributes, options?: { ignoreFalseyValues?: boolean}) => TData}
+   */
+  let path = [];
+  const findAllAndReplace = (data, attributes, { ignoreFalseyValues = false } = {}) => {
+    return Object.entries(attributes).reduce(
+      (acc, [key, value]) => {
+        if (
+          ignoreFalseyValues &&
+          (acc === null || acc === undefined || acc[key] === undefined || acc[key] === null)
+        ) {
+          return acc;
+        }
+
+        if (predicate(value, { path: [...path, key], parent: acc })) {
+          path = [...path, key];
+          acc[key] =
+            typeof replacement === 'function'
+              ? replacement(acc[key], { path, parent: acc })
+              : replacement;
+          path = [];
```

Comment:
I find it a bit confusing on why you assign path to empty it later on here and in the following two instances.

What do you think about:

```suggestion
          path = ;
            typeof replacement === 'function'
              ? replacement(acc[key], { path: [...path, key], parent: acc })
              : replacement;
          path = [];
```

I'm wondering if this would allow you to get rid of the global `path` variable entirely?

Reply:
You need to have the global path because when you dive down one layer (for components / dynamic zones etc) you need that path to be remained although this might be able to be part of the func...

Nice idea, i'll give it a look.

Reply:
looks like you were on to something!